lib/datadog/ci/configuration/components.rb

```diff
@@ -16,15 +17,56 @@ def initialize(settings)
         end
 
         def activate_ci!(settings)
+          agentless_transport = nil
+
+          if settings.ci.agentless_mode_enabled
+            if settings.api_key.nil?
+              # agentless mode is requested but no API key is provided -
+              # we cannot continue and log an error
+              # Tests are running without CI visibility enabled
+
+              Datadog.logger.error(
+                "DATADOG CONFIGURATION - CI VISIBILITY - ATTENTION - " \
+                "Agentless mode was enabled but DD_API_KEY is not set: CI visibility is disabled. " \
+                "Please make sure to set valid api key in DD_API_KEY environment variable"
+              )
+
+              settings.ci.enabled = false
+              return
+            else
+              dd_site = settings.site || "datadoghq.com"
+              agentless_url = settings.ci.agentless_url ||
+                "https://#{Ext::Transport::TEST_VISIBILITY_INTAKE_HOST_PREFIX}.#{dd_site}:443"
+
+              agentless_transport = Datadog::CI::TestVisibility::Transport.new(
+                api_key: settings.api_key,
+                url: agentless_url,
+                env: settings.env
+              )
+            end
+          end
```

Comment:
style nitpick: Do you think this method would be clearer if this block of code was in its own method? And then in `activate_ci!` you could just have `agentless_transport = build_agentless_transport(settings)` or whatever name you think makes the most sense?